debezium-core/src/main/java/io/debezium/pipeline/spi/OffsetContext.java

```diff
@@ -27,7 +27,7 @@
         OffsetContext load(Map<String, ?> offset);
     }
 
-    Map<String, ?> getPartition();
+    Map<String, String> getPartition();
```

Comment:
@gunnarmorling Out of curiosity? Why did you have narrowed the type?

Reply:
`Heartbeat.create()` creates the heartbeat facility. IIRC the `OffsetPosition` is created lazily for performance reasons. You'd need to pass the position when you call `heartbeat` method but the positions is factually necessary only very rarely - if the heartbeat message has to be created and delivered.